### PR TITLE
Recipe App

### PR DESCRIPTION
**Pull request by phelian23**
In this PR we Create:

- A login page.
- A registration page.
- A food list (with all CRUD implementation, except for 'update').
- A recipes list (with all CRUD implementation, except for 'update').
- A general shopping list view (all missing food for all your recipes and total price).
- A list of all public recipes from other users with their names and total prices.
- Inventories list (with all CRUD implementation, except for 'update').
- Inventory shopping list, a shopping list, but only taking into consideration a chosen recipe and inventory.

### CODE REVIEW

File: app/models/inventory.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Inventory < ApplicationRecord
+  belongs_to :user
+  has_many :inventory_foods
+  has_many :foods
```
Comment (MrBrN197) [inline app/models/inventory.rb:4]: - [ ] according to the diagram and the migrations you've set up an `Inventory` model should not have many foods. what you can do is say that it has many foods through the inventory_foods table.
```suggestion
  has_many :foods, through: :inventory_foods
```

File: app/models/recipe.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Recipe < ApplicationRecord
+  belongs_to :user
+  has_many :recipe_foods
+  has_many :foods
```
Comment (MrBrN197) [inline app/models/recipe.rb:4]: - [ ] According to your migration and the ERD diagram. a Recipe should not have a direct relationship to the "foods" table. you can use the recipe_foods to connect the recipe and foods.

```suggestion
  has_many :foods, through: :recipe_foods
```

**Review by MrBrN197 — CHANGES_REQUESTED** [review]
## Hi,  @aakbarkhan , @RolandM99  & @phelian23 👋 

You've done a really nice job so far.
There are some issues that you still need to work on to go to the next project but you are almost there!
## To highlight 👏👏
- ✔migrations setup correctly.
- ✔ linter checks passing

## Required Changes ♻️
_Check the comments under the review._

- [ ] please ensure to add a Navbar that can link to all the other pages, for example with links to  recipes, foods, and the public recipes e.t.c.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by descholar-ceo — APPROVED** [review]
Hi @phelian23 && @RolandM99 && @aakbarkhan , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**